src/server/views/dashboard/queueJobsByState.js

```diff
@@ -34,12 +34,27 @@ async function _json(req, res) {
 
   if (!isValidState(state, queue.IS_BEE)) return res.status(400).json({ message: `Invalid state requested: ${state}` });
 
+  const page = parseInt(req.query.page, 10) || 1;
+  const pageSize = parseInt(req.query.pageSize, 10) || 1000;
+
+  const startId = (page - 1) * pageSize;
+  const endId = startId + pageSize - 1;
+
   let jobs;
   if (queue.IS_BEE) {
-    jobs = await queue.getJobs(state, { size: 1000 });
+    const page = {};
+
+    if (['failed', 'succeeded'].includes(state)) {
+      page.size = pageSize;
+    } else {
+      page.start = startId;
+      page.end = endId;
+    }
```

Comment:
This would be a bit terser, and maybe preferable.
```suggestion
    const page = ['failed', 'succeeded'].includes(state) ? { size: pageSize } : { start: startId, end: endId };
```